pkg/scheduler/objects/ask_events.go

```diff
@@ -0,0 +1,93 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package objects
+
+import (
+	"fmt"
+	"time"
+
+	"golang.org/x/time/rate"
+
+	"github.com/apache/yunikorn-core/pkg/common/resources"
+	"github.com/apache/yunikorn-core/pkg/events"
+)
+
+// Ask-specific events. These events are of REQUEST type, so they are eventually sent to the respective pods in K8s.
+type askEvents struct {
+	eventSystem events.EventSystem
+	ask         *AllocationAsk
+	limiter     *rate.Limiter
+}
+
+func (ae *askEvents) sendRequestDoesNotFitInQueue(headroom *resources.Resource, queuePath string) {
```

Comment:
Can we rename this to align with other method names? ex, sendRequestExceeds..
or we can follow FitIn or DoesNotFitIn for all methods.

Reply:
Done